GVFS/GVFS.Common/Git/GitObjects.cs

```diff
@@ -594,6 +594,8 @@ private Stream OpenTempLooseObjectStream(string path)
 
         private LooseObjectToWrite GetLooseObjectDestination(string sha)
         {
+            // Ensure SHA path is lowercase for case-sensitive filesystems
+            sha = sha.ToLower();

```

Comment:
Same comment here as in `GitRepo. GetLooseBlobState`, can we avoid always calling `ToLower()` and make sure the caller is passing in the correct casing?